nano/node/online_reps.cpp

```diff
@@ -1,101 +1,131 @@
+#include <nano/node/nodeconfig.hpp>
 #include <nano/node/online_reps.hpp>
 #include <nano/secure/blockstore.hpp>
-#include <nano/secure/common.hpp>
 #include <nano/secure/ledger.hpp>
 
-nano::online_reps::online_reps (nano::ledger & ledger_a, nano::network_params & network_params_a, nano::uint128_t minimum_a) :
-ledger (ledger_a),
-network_params (network_params_a),
-minimum (minimum_a)
+nano::online_reps::online_reps (nano::ledger & ledger_a, nano::node_config const & config_a) :
+ledger{ ledger_a },
+config{ config_a }
 {
 	if (!ledger.store.init_error ())
 	{
 		auto transaction (ledger.store.tx_begin_read ());
-		online = trend (transaction);
+		trended_m = calculate_trend (transaction);
 	}
 }
 
 void nano::online_reps::observe (nano::account const & rep_a)
 {
 	if (ledger.weight (rep_a) > 0)
 	{
+		auto now = std::chrono::steady_clock::now ();
 		nano::lock_guard<std::mutex> lock (mutex);
-		reps.insert (rep_a);
+		auto new_insert = reps.get<tag_account> ().erase (rep_a) == 0;
+		reps.insert ({ now, rep_a });
```

Comment:
Should `now` be captured inside the critical section in case it takes a while to grab the mutex?

Reply:
Can use emplace with result check as here:
https://github.com/nanocurrency/nano-node/blob/develop/nano/node/node.cpp#L1269
```
Effects: Inserts a value_type object constructed with std::forward<Args>(args)... into the multi_index_container to which the index belongs if
* the index is non-unique OR no other element exists with equivalent key,
* AND insertion is allowed by all other indices of the multi_index_container.
Returns: The return value is a pair p. p.second is true if and only if insertion took place.
```

Reply:
Emplace can't be used with POD type initialization.